### DISCUSSION TITLE
UPS how to set UnitOfMeasurement?

### DISCUSSION THREAD

**Discussion by ebaj-tech** [alexeybusygin/ShippingRates#67] [Q&A] [answered]
After calling GetRatesAsync(), I receive the following error message: **111057 Invalid Unit of Measurement for the given country**

How can the UnitOfMeasurement for packages be set? The following instructions should be possible: **shipment.ShipmentTotalWeight.UnitOfMeasurement.Code = "KGS";
shipment.ShipmentTotalWeight.UnitOfMeasurement.Description = "KILOGRAMS";
shipment.Package.Dimensions.UnitOfMeasurement.Code = "CM";
shipment.Package.Dimensions.UnitOfMeasurement.Description = "CENTIMETERS";**

Or when creating the packages: **var package = new Package( 12, 12, 12, 35, 150, "CM", "KGS" );**

How can the error be fixed, or how can the correct units be used?

**Comment by alexeybusygin** [comment]
What country do you ship from/to in the request?

**Reply by ebaj-tech** [reply]
Is there a way in the current version to set the UnitOfMeasurement for Germany and all other countries that use the metric system?

**Reply by alexeybusygin** [reply]
In the current version, the UnitOfMeasurement cannot be set. I created issue #69 for this fix.

**Comment by alexeybusygin** [comment, marked as answer]
Starting from v2.2.0, the UPS provider sends kg/cm dimensions for shipments from outside the US. Package dimensions are automatically converted to the necessary system of measurement. Along with the existing Packages class (accepting values for lbs and inches), a new PackageKgCm class is introduced for dimensions in kg and cm. Both classes can be used for shipments.
